### PR TITLE
added the  test for evidence.py

### PR DESCRIPTION
**Pull request by Sekhar-Kumar-Dash**
## Fixes Issue #770 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by AlyaGomaa** [comment]
hey @Sekhar-Kumar-Dash can you check why they're failing?

File: tests/module_factory.py [hunk 1]
```diff
@@ -353,3 +368,64 @@ def create_set_evidence_helper(self, mock_db):
         """Create an instance of SetEvidenceHelper."""
         set_evidence_helper = SetEvidnceHelper(mock_db)
         return set_evidence_helper
+
+    def create_attacker_obj(value="192.168.1.1", direction=Direction.SRC, attacker_type=IoCType.IP):
```
Comment (AlyaGomaa) [inline tests/module_factory.py:372]: All the added functions are missing "self" as first parameter

**Review by AlyaGomaa — CHANGES_REQUESTED** [review]
hey @Sekhar-Kumar-Dash can you fix this and update PR with the latest develop so i can see why CI is failing?

**Comment by Sekhar-Kumar-Dash** [comment]
> hey @Sekhar-Kumar-Dash can you fix this and update PR with the latest develop so i can see why CI is failing?

hey @AlyaGomaa  okay i will make the required changes and then i will update the PR

**Comment by AlyaGomaa** [comment]
hey @Sekhar-Kumar-Dash for some reason, the changes you made to evidence.py is what's causing CI to fail, i reverted them to the latest develop and will see how it goes

**Pull request merged**